src/backend/distributed/worker/worker_create_or_replace.c

```diff
@@ -162,6 +195,68 @@ GenerateBackupNameForCollision(const ObjectAddress *address)
 }
 
 
+/*
+ * CreateRenameTypeStmt creates a rename statement for a type based on its ObjectAddress.
+ * The rename statement will rename the existing object on its address to the value
+ * provided in newName.
+ */
+static RenameStmt *
+CreateRenameTypeStmt(const ObjectAddress *address, char *newName)
+{
+	RenameStmt *stmt = makeNode(RenameStmt);
+
+	stmt->renameType = OBJECT_TYPE;
+	stmt->object = (Node *) stringToQualifiedNameList(format_type_be_qualified(
+														  address->objectId));
+	stmt->newname = newName;
+
+	return stmt;
+}
+
+
+/*
+ * CreateRenameTypeStmt creates a rename statement for a type based on its ObjectAddress.
+ * The rename statement will rename the existing object on its address to the value
+ * provided in newName.
+ */
+static RenameStmt *
+CreateRenameProcStmt(const ObjectAddress *address, char *newName)
```

Comment:
would be good to have code coverage for this function

Reply:
Coverage here will always be a little spotty because it'll only ever be called for aggregates in PG11. Will add a test case for that scenario tho